src/main/scala/is/hail/expr/HailRep.scala

```diff
@@ -2,8 +2,9 @@ package is.hail.expr
 
 import is.hail.utils.Interval
 import is.hail.variant.{AltAllele, Call, Genotype, Locus, Variant}
+import scala.collection.mutable
 
-trait HailRep[T] { self =>
+trait HailRep[-T] { self =>
```

Comment:
The way I think about it: anywhere I need a hail version of some Scala type `T`, I can just as well use the hail version of any super type of `T` (it's less specific, but still correct).

If `HailRep` is covariant, then I can use `HailRep[mutable.WrappedArray[T]]` whenever I need a `HailRep[Seq[T]]`. This is wrong, because a `Set[T]` should have a hail representation as a `TSet` not a `TArray`.

I actually prefer the invariant situation, with specific implicits for specific classes (e.g. `mutable.WrappedArray`), because if Hail adds sub/super-typing to its type system, then Scala will not be able to resolve this situation:

```
implicit object hrSeq extends HailRep[Seq[T]] { /* ... */ }
implicit object hrArray extends HailRep[mutable.WrappedArray[T]] { /* ... */ }

val x = implicitly[HailRep[mutable.WrappedArray[T]]
```

it will be unable to choose between `hrSeq` and `hrArray` because both *may flow to* (i.e. are sub types of) `HailRep[mutable.WrappedArray[T]]`.